api/src/pcapi/scripts/add_venue_providers_on_venue_with_siret/main.py

```diff
@@ -0,0 +1,122 @@
+Assumed path to the script (copy-paste in github actions):
+
+https://github.com/pass-culture/pass-culture-main/blob/ogeber/pc-36399-add-venue-providers-on-venue-with-siret/api/src/pcapi/scripts/add_venue_providers_on_venue_with_siret/main.py
+
+"""
+
+import argparse
+import csv
+import logging
+import os
+
+import sqlalchemy as sa
+import sqlalchemy.exc as sa_exc
+import sqlalchemy.orm as sa_orm
+
+from pcapi.app import app
+from pcapi.core.history import api as history_api
+from pcapi.core.history import models as history_models
+from pcapi.core.offerers.models import Venue
+from pcapi.core.providers.models import VenueProvider
+from pcapi.models import db
+from pcapi.repository.session_management import atomic
+from pcapi.repository.session_management import mark_transaction_as_invalid
+
+
+logger = logging.getLogger(__name__)
+
+VENUE_ID_HEADER = "venue_id"
+
+
+def _write_modifications(modifications: list[tuple[int, str]], filename: str) -> None:
+    # csv output to check what has been done and what failed
+    output_file = f"{os.environ['OUTPUT_DIRECTORY']}/{filename}"
+    logger.info("Exporting data to %s", output_file)
+
+    with open(output_file, "w", encoding="utf-8") as csvfile:
+        writer = csv.writer(csvfile, dialect=csv.excel, delimiter=";", quoting=csv.QUOTE_NONNUMERIC)
+        writer.writerow([VENUE_ID_HEADER, "Modification"])
+        writer.writerows(modifications)
+
+
+@atomic()
+def main(not_dry: bool) -> None:
+    log_modifications: list[tuple[int, str]] = []
+    log_fails: list[tuple[int, str]] = []
+    venues_without_siret_with_providers: list[Venue] = (
+        db.session.query(Venue)
+        .join(Venue.venueProviders)
+        .filter(
+            Venue.siret.is_(None),
+            sa.exists().where(VenueProvider.venueId == Venue.id).correlate(Venue),
+        )
+        .options(
+            sa_orm.load_only(Venue.id, Venue.siret, Venue.managingOffererId),
+            sa_orm.joinedload(Venue.venueProviders).load_only(
+                VenueProvider.venueId,
+                VenueProvider.providerId,
+            ),
+        )
+        .all()
+    )
+
+    venues_with_siret_by_offerer_query = (
+        db.session.query(sa.func.min(Venue.id))
+        .filter(Venue.siret.is_not(None))
+        # .options(sa_orm.load_only(Venue.managingOffererId))
+        .having(sa.func.count(Venue.id) == 1)
+        .group_by(Venue.managingOffererId)
+    )
+
+    for venue in venues_without_siret_with_providers:
+        venue_id_with_siret = venues_with_siret_by_offerer_query.filter(
+            Venue.managingOffererId == venue.managingOffererId
+        ).scalar()
+
+        new_providers: list[VenueProvider] = [
+            VenueProvider(venueId=venue_id_with_siret, providerId=venue_provider.providerId)
+            for venue_provider in venue.venueProviders
+        ]
+        print(f"Venue {venue.id} will be synch. with those providers {new_providers}")
```

Comment:
C'est plutôt cela non ?

```suggestion
        print(f"Venue {venue.id} will be synch. with those providers {vp.provider for vp in new_providers}")
```